src/TodoListItem.js

```diff
@@ -12,14 +15,33 @@ class TodoListItem extends React.Component {
   }
 
   handleChange = () => {
-    let { onUpdateItem, id, item } = this.props
+    let { onUpdateItem, item } = this.props
+    let { id } = this.props.item
+    // console.log(this.props);
+    console.log('this.props', this.props);
+
     item.complete = !item.complete
 
-    this.setState({ complete: item.complete }, () => {
-      setTimeout(() => {
-        onUpdateItem(id, item)
-      }, 300)
+    console.log("id", id);
+    fetch(`http://localhost:5000/complete/${id}`, {
+      method: 'put'
     })
+    // .then(results => {
+    //   console.log(results);
+    // })
+    .then( response => response.json() )
+      .then(results => {
+        this.setState({ complete: results }, () => {
+          setTimeout(() => {
```

Comment:
Why are you using a set timeout here?
Im pretty sure you can just set the state, then call onUpdateItem().
Also what exactly is onUpdateItem doing here?